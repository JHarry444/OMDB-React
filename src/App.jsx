import 'bootstrap/dist/css/bootstrap.min.css';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Navigation from './components/Navigation';
import Search from './components/Search';
import Details from './components/Details';

const App = () => (
  <Router>
    <Navigation />
    <Switch>
      <Route exact path="/">
        <Search />
      </Route>
      <Route path="/details/:id">
        <Details />
      </Route>
    </Switch>
  </Router>
);

export default App;
