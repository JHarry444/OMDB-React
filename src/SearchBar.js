import React from 'react';
import { InputGroup, FormControl, Button, Form } from 'react-bootstrap';

export default function SearchBar(props) {
    return (
        <Form onSubmit={props.searchFunc}>
            <InputGroup>
                <FormControl onChange={props.handleChange} value={props.SearchTerm}
                    placeholder="Enter search term here"
                />
                <InputGroup.Append>
                    <Button onClick={props.searchFunc}>Button</Button>
                </InputGroup.Append>
            </InputGroup>
        </Form>
    );
}