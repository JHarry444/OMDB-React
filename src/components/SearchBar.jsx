import {
  InputGroup, FormControl, Button, Form,
} from 'react-bootstrap';
import PropTypes from 'prop-types';

const SearchBar = ({ searchFunc, handleChange, searchTerm }) => (
  <Form onSubmit={searchFunc}>
    <InputGroup>
      <FormControl
        onChange={handleChange}
        value={searchTerm}
        placeholder="Enter search term here"
      />
      <InputGroup.Append>
        <Button onClick={searchFunc}>Button</Button>
      </InputGroup.Append>
    </InputGroup>
  </Form>
);

SearchBar.propTypes = {
  searchFunc: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  searchTerm: PropTypes.string.isRequired,
};

export default SearchBar;
