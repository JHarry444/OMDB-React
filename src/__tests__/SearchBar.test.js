import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';
import renderer from 'react-test-renderer';
import SearchBar from '../SearchBar';


it('renders without crashing', () => {
  shallow(<SearchBar />);
});

it('matches snapshot', () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains query', () => {
  // const searchBar = mount(<SearchBar searchTerm='bloop' />)
  // expect(searchBar.find('input[type="text"]').text()).toEqual('bloop');
})

