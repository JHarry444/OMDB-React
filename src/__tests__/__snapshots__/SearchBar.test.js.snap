// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className=""
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      placeholder="Enter search term here"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="button"
      >
        Button
      </button>
    </div>
  </div>
</form>
`;
